tests/unit/src/kakarot/precompiles/test_sha256.py

```diff
@@ -0,0 +1,46 @@
+import random
+from datetime import timedelta
+from hashlib import sha256 as py_sha256
+
+import pytest
+import pytest_asyncio
+from hypothesis import given, settings
+from hypothesis.strategies import integers
+from starkware.starknet.testing.starknet import Starknet
+
+
+@pytest_asyncio.fixture(scope="module")
+async def sha256(starknet: Starknet):
+    return await starknet.deploy(
+        source="./tests/unit/src/kakarot/precompiles/test_sha256.cairo",
+        cairo_path=["src"],
+        disable_hint_validation=True,
+    )
+
+
+@pytest.mark.asyncio
+@pytest.mark.SHA256
+class TestSHA256:
+    @given(
+        data_len=integers(min_value=1, max_value=56),
+    )
+    @settings(deadline=timedelta(milliseconds=20000), max_examples=10)
+    async def test_sha256_should_return_correct_hash(self, sha256, data_len):
+        # Set seed
+        random.seed(0)
```

Comment:
I think that we don't care in this specific example but setting the `seed` to be always the same in a parametrized test could lead to unexpected under testing as all cases may become the same...
funny to see how the `randbytes` works actually
<img width="303" alt="image" src="https://user-images.githubusercontent.com/18620296/215038662-666913c9-28a3-4467-8c5a-9ef06eb4dd74.png">

Reply:
That's funny to see its behavior actually 😅 and yeah setting the seed here kind of defeats the purpose of doing using Hypothesis because it would lead to the same test cases, let me remove it 👍